### PR TITLE
feat: Razorpay subscription for Non profit memberships

### PR DESCRIPTION
**Pull request by scmmishra**
This PR makes the following changes
1. Integrates razorpay with Non Profit Membership
1. Added membership settings DocType
1. Added webhook for subscription activated and charged

### CODE REVIEW

File: erpnext/non_profit/doctype/membership_settings/membership_settings.json [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  {
+   "depends_on": "eval:doc.enable_razorpay",
+   "fieldname": "razorpay_settings_section",
+   "fieldtype": "Section Break",
+   "label": "RazorPay Settings"
+  },
+  {
+   "description": "The number of billing cycles for which the customer should be charged. For example, if a customer is buying a 1-year membership that should be billed on a monthly basis, this value should be 12.",
+   "fieldname": "billing_frequency",
+   "fieldtype": "Data",
```
Comment (surajshetty3416) [inline erpnext/non_profit/doctype/membership_settings/membership_settings.json]:
```suggestion
   "fieldtype": "Int",
```

File: erpnext/non_profit/doctype/membership/membership.py [hunk 2]
```diff
@@ -43,11 +47,95 @@ def validate(self):
...
+
+
+
+
+
+
+
+
+
+
```
Comment (surajshetty3416) [inline erpnext/non_profit/doctype/membership/membership.py]: delete...

File: erpnext/non_profit/doctype/member/member.py [hunk 3]
```diff
@@ -14,8 +17,114 @@ def onload(self):
...
+	Returns:
+	    Dictionary: Dictionary with subscription details
+	    {
+	    	'subscription_details': {
+	    								'plan_id': 'plan_EXwyxDYDCj3X4v',
+							  			'billing_frequency': 24,
+							  			'customer_notify': 1
+							  		},
+			'subscription_id': 'sub_EZycCvXFvqnC6p'
+		}
```
Comment (surajshetty3416) [inline erpnext/non_profit/doctype/member/member.py]: spaces and tabs...
Screenshot 2020-04-13 at 2 33 18 PM

File: erpnext/non_profit/doctype/membership_settings/membership_settings.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+// Copyright (c) 2020, Frappe Technologies Pvt. Ltd. and contributors
+// For license information, please see license.txt
+
+frappe.ui.form.on('Membership Settings', {
+	refresh: function(frm) {
+		// if (frm.doc.enable_razorpay) {
+		// 	frm.add_custom_button(__("Fetch Plans from RazorPay"), () => {
+		// 		frm.trigger("fetch_razorpay_plans")
+		// 	});
+		// }
```
Comment (surajshetty3416) [inline erpnext/non_profit/doctype/membership_settings/membership_settings.js]: remove commented code if not required.

**Review by surajshetty3416 — CHANGES_REQUESTED** [review]

**Comment by scmmishra** [comment]
@nabinhait can you review this?

**Pull request merged**